ci/build-src-tarball.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -ex
+
+# Determine the name of the tarball
+tag=dev
+if [[ $GITHUB_REF == refs/tags/v* ]]; then
+  tag=${GITHUB_REF:10}
+fi
+pkgname=wasmtime-$tag-src
+rm -rf /tmp/$pkgname
+mkdir /tmp/$pkgname
+
+# Vendor all crates.io dependencies since this is supposed to be an
+# offline-only-compatible tarball
+mkdir /tmp/$pkgname/.cargo
+cargo vendor > /tmp/$pkgname/.cargo/config.toml
```

Comment:
I think I'll leave this as the default options since that's primarily only intended if it needs to be opted-in to, but I don't think there would be any harm in passing it in.